### PR TITLE
[CICD-221]Refactor entrypoint.sh

### PR DESCRIPTION
**Pull request by marksabbath**
# JIRA Ticket

[CICD-221](https://wpengine.atlassian.net/browse/CICD-221)

## What Are We Doing Here

This PR introduces a more organized `entrypoint.sh` with functions and a few fixes. It also introduces a new variable labeled `CICD_VENDOR` so we can internally set them on each CI/CD platform and use it to track which systems are in use in the future.

### CODE REVIEW

**Comment by marksabbath** [comment]
> Thanks for the updates!
> 
> Just did a another test and found this error if I run without setting a value for `SRC_PATH` to let it take the default:
> 
> ```
> ...
> !!! MULTIPLEX SSH CONNECTION ESTABLISHED !!!
> sending incremental file list
> rsync: [sender] link_stat "/site/'.'" failed: No such file or directory (2)
> ```

Should be fixed!

**Review by apmatthews — DISMISSED** [review]

File: .changeset/swift-maps-grab.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+"@wpengine/site-deploy": major
+---
+
+# Refactor the Main Script
+
+These updates the main script to be more generic and allow the script and image to be used around other CI/CD Vendors.
```
Comment (apmatthews) [inline .changeset/swift-maps-grab.md:7]:
```suggestion
Updates the main script to be more generic and allow the script and image to be used around other CI/CD Vendors.
```

File: .changeset/swift-maps-grab.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+In order to use this script, each CI/CD vendor will need to set the environment variables accordingly:
+
+```sh
+REMOTE_PATH # Defualt is empty
```
Comment (apmatthews) [inline .changeset/swift-maps-grab.md:12]:
```suggestion
REMOTE_PATH # Default is empty
```

File: .changeset/swift-maps-grab.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+SRC_PATH # Default is the current directory
+FLAGS # Default is -azvr --inplace --exclude=".*"
+PHP_LINT # Default is "FALSE"
+CACHE_CLEAR # Default is "TRUE"
+SCRIPT # Default is empty
+CICD_VENDOR= # Default is "wpe-cicd"
```
Comment (apmatthews) [inline .changeset/swift-maps-grab.md:18]:
```suggestion
CICD_VENDOR # Default is "wpe-cicd"
```

**Review by apmatthews — APPROVED** [review]

**Pull request merged**